src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -706,11 +706,23 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
 
             writer.WriteStartObject();
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
             {
+                writer.WritePropertyName(JwtPayloadUtf8Bytes.Aud);
+                writer.WriteStartArray();
+                foreach (string audience in tokenDescriptor.Audiences) { writer.WriteStringValue(audience); }
+
+                if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
+                    writer.WriteStringValue(tokenDescriptor.Audience);
+
+                writer.WriteEndArray();
                 audienceSet = true;
+            }
+            else if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
```

Comment:
with this else if, when the user set Audience and Audiences, Audience will not be written.
Is that what we want?

Reply:
I am not sure i am understanding.
The OpenIdConnectConfigurationSerializer writes out an array of strings.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b4560536c4abde119272bd173ea530e0211e26c3/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs#L482

Reply:
Fixed it so both will be written, without duplication. I'm not a fan of [overloading the WriteStrings method the way I did](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b9497bcbf17977ddda036eecf92aaeef33c9faf1/src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs#L1192) but it seemed the neatest way to combine them without altering the audiences collection or allocating a new collection. Open to other ideas though.